src/utils/input_parser.cpp

```diff
@@ -122,6 +122,21 @@ inline Duration get_priority(const rapidjson::Value& object) {
   return priority;
 }
 
+inline size_t get_max_number_of_tasks(const rapidjson::Value& object) {
+  size_t max_number_of_tasks = std::numeric_limits<size_t>::max();
+  if (object.HasMember("max_number_of_tasks")) {
+    if (!object["max_number_of_tasks"].IsUint()) {
+      throw Exception(ERROR::INPUT, "Invalid max_number_of_tasks value.");
+    }
+    max_number_of_tasks = object["max_number_of_tasks"].GetUint();
+    if (!max_number_of_tasks) {
+      throw Exception(ERROR::INPUT, "Invalid max_number_of_tasks value.");
+    }
+    
```

Comment:
I'm curious about when this would be triggered? For example in the similar `get_service`, we don't have that check at all.

Reply:
In my honest opinion, `max_number_of_tasks` should be a positive number or not provided

Reply:
Agreed, my point is that we already check that by throwing if `max_number_of_tasks` is provided and `object["max_number_of_tasks"].IsUint()` is false.

Reply:
Ok, didn't know. I will remove it !

Reply:
Removed !